common/component/azure/servicebus/message_pubsub.go

```diff
@@ -58,6 +59,12 @@ func addMessageAttributesToMetadata(metadata map[string]string, asbMsg *azservic
 		metadata = map[string]string{}
 	}
 
+	for key, val := range asbMsg.ApplicationProperties {
+		metadata["metadata."+key] = fmt.Sprintf("%v", val)
```

Comment:
Can we avoid `fmt.Sprintf` in a hot path? Worst case, how about using [`cast.ToString()`](https://pkg.go.dev/github.com/spf13/cast#ToString) which at least avoids some allocations?

Reply:
Done.